float64 reference
float64 actuator
float64 error
float64 i_error
float64 d_error